src/middleware.ts

```diff
@@ -44,7 +44,7 @@ const createRouterMiddleware = (structure: Structure) => (middlewareOpts: Router
     if (Router && !isServer && type !== LOCATION_CHANGE) {
       const storeLocation = structure.getIn(store.getState(), [reducerKey, 'location']) as LocationState
       if (Router.asPath !== storeLocation.href) {
-        next(onLocationChanged(locationFromUrl(Router.asPath)))
+        return next(onLocationChanged(locationFromUrl(Router.asPath)))
```

Comment:
@laooola We probably don't want to return here as it may prevent the action from reaching out the other middlewares.

Reply:
I think you're right; if the action is e.g. a `CALL_ROUTER_METHOD` action, we would not further process it in case we have a location mismatch here. IMHO we should use `store.dispatch(onLocationChanged(locationFromUrl(Router.asPath)))` without the return to dispatch the action. If we do it this way the action will be processed by all middlewares in the chain. If we use `next()` only the subsequent middlewares in the chain will actually receive the action (unless the router middleware is the first in the chain).